### PR TITLE
upgrade jda

### PR DESCRIPTION
**Pull request by Taz03**
- upgrade jda
- added new event forwards
- updated code according to API changes
- update `UserInteractor` interface to have empty default implementation
- cat
image

### CODE REVIEW

**Comment by Taz03** [comment]
need to merge #1016 to fix deprecated warnings

File: application/src/test/java/org/togetherjava/tjbot/jda/JdaTester.java [hunk 1]
```diff
@@ -20,9 +19,7 @@
 import net.dv8tion.jda.api.interactions.callbacks.IReplyCallback;
 import net.dv8tion.jda.api.interactions.components.ItemComponent;
 import net.dv8tion.jda.api.interactions.components.LayoutComponent;
-import net.dv8tion.jda.api.requests.ErrorResponse;
-import net.dv8tion.jda.api.requests.Response;
-import net.dv8tion.jda.api.requests.RestAction;
+import net.dv8tion.jda.api.requests.*;
```
Comment (SquidXTV) [inline application/src/test/java/org/togetherjava/tjbot/jda/JdaTester.java:22]: No star imports please

**Review by SquidXTV — CHANGES_REQUESTED** [review]

File: application/src/test/java/org/togetherjava/tjbot/jda/JdaTester.java (hunk 1 above)
Comment (Taz03) [reply application/src/test/java/org/togetherjava/tjbot/jda/JdaTester.java:22]: actually ig spotless did that, there are star imports in the codebase already

Comment (SquidXTV) [reply application/src/test/java/org/togetherjava/tjbot/jda/JdaTester.java:22]: It is probably not spotless but intellij

**Review by christolis — APPROVED** [review]

**Review by SquidXTV — APPROVED** [review]

**Pull request merged**